src/server/dflycmd.h

```diff
@@ -101,16 +101,44 @@ class DflyCmd {
 
   // Stores information related to a single replica.
   struct ReplicaInfo {
-    ReplicaInfo(unsigned flow_count, Context::ErrHandler err_handler)
-        : state{SyncState::PREPARATION}, cntx{std::move(err_handler)}, flows{flow_count} {
+    ReplicaInfo(unsigned flow_count, std::string address, std::string port,
+                Context::ErrHandler err_handler)
+        : state{SyncState::PREPARATION}, address{address}, port{port}, cntx{std::move(err_handler)},
+          flows{flow_count} {
     }
 
     SyncState state;
+    std::string address;
+    std::string port;
     Context cntx;
 
     std::vector<FlowInfo> flows;
     ::boost::fibers::mutex mu;  // See top of header for locking levels.
   };
+  struct ReplicaData {
```

Comment:
Since we already have ReplicaInfo, ReplicaData is a bit confusing. Maybe ReplicaRoleInfo?